src/fen_util.rs

```diff
@@ -40,15 +54,30 @@ pub fn place_pieces(chessboard: &mut Chessboard, fen_rows: &str) {
     }
 }
 
-/// Function to check if a given FEN (Forsyth-Edwards Notation) string is valid
+/// Validates a Forsyth-Edwards Notation (FEN) string to check if it conforms to the standard format.
 ///
 /// # Arguments
 ///
-/// * `fen` - A FEN string to be validated.
+/// - `fen`: A FEN string to be validated.
 ///
 /// # Returns
 ///
 /// A boolean indicating whether the FEN string is valid.
+///
+/// # Example
+///
+/// ```
+/// use kn_o3::fen_util::valid_fen;
+///
+/// let valid_fen_string = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1";
+/// assert!(valid_fen(valid_fen_string));
+///
+/// let invalid_fen_string = "invalid_fen_string";
+/// assert!(!valid_fen(invalid_fen_string));
+/// ```
+///
+/// The function checks if a given FEN string is valid based on the standard FEN format rules.
+/// It verifies the piece placement, active color, castling rights, en passant square, and move counters.
```

Comment:
this function is also pretty self explanatory. The example seems overkill to me.